src/api/index.tsx

```diff
@@ -293,26 +293,6 @@ type CommentRequest = {
   lineno?: number | null;
 };
 
-type CreateOrUpdateCommentParamsBase = {
```

Comment:
I saw in your comment that you had to remove this, but I'm not sure I understand why. AFAIU this *should* work. What exactly was the problem with it?

Reply:
It's only possible if I switch to [discriminated unions](https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html) (i.e. each type will need a unique `type` field) and I don't think that's worth it.

Reply:
I didn't notice this before because I guess I didn't have any code that tried to call the function in all possible ways.